packages/headless/src/app/headless-engine.ts

```diff
@@ -88,6 +93,15 @@ export interface HeadlessConfigurationOptions
   };
 }
 
+export type Engine<State extends object = SearchAppState> = CoreEngine<
```

Comment:
If `Engine<State>` is meant to be used by developers, it should have a second generic parameter for the `ThunkExtraArguments` type.

Unless I am mistaken `Engine<State>` only works for search right now.

Reply:
Yep, the `Engine` is tied to the search api. I initially wanted to use `Engine` as the base interface, and expose a new `SearchEngine` interface. Unfortunately, this would be breaking.

Instead, `Engine` continues to be tied to the search API, and `CoreEngine` is the base interface that can be extended by different use-cases. It's possible to specify the second generic param on `CoreEngine`.

Reply:
I thought maybe you could introduce the `SearchEngine` interface in a non-breaking way and deprecate `Engine<State>`.

Reply:
Yep, me too. Unfortunately, the `Engine` interface is publicly exposed. We could look into deprecating and replacing as part of v1.